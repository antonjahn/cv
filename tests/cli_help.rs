//! CLI integration tests for help printout using insta

use assert_cmd::Command;
use insta::assert_snapshot;

#[test]
fn cli_help_printout() {
    let mut cmd = Command::cargo_bin("cv").unwrap();
    cmd.arg("--help");
    let output = cmd.assert().success().get_output().stdout.clone();
    let help = String::from_utf8_lossy(&output);
    assert_snapshot!(help);
}

#[test]
fn cli_zig_help_printout() {
    let mut cmd = Command::cargo_bin("cv").unwrap();
    cmd.args(["zig", "--help"]);
    let output = cmd.assert().success().get_output().stdout.clone();
    let help = String::from_utf8_lossy(&output);
    assert_snapshot!(help);
}

#[test]
fn cli_zig_list_prints_versions() {
    let mut cmd = Command::cargo_bin("cv").unwrap();
    cmd.args(["zig", "list"]);
    let output = cmd.assert().success().get_output().stdout.clone();
    let out = String::from_utf8_lossy(&output);
    // Count lines that look like version output
    let version_lines = out.lines().filter(|l| l.contains("zig-")).count();
    assert!(
        version_lines >= 3,
        "Expected at least 3 zig versions, got {}. Output: {}",
        version_lines,
        out
    );
}
