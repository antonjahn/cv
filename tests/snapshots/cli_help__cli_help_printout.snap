---
source: tests/cli_help.rs
expression: help
---
A fast, minimal C/C++ toolchain manager.

Usage: cv [COMMAND]

Commands:
  zig      Manage zig versions and installations
  version  Display cv's version
  help     Print this message or the help of the given subcommand(s)

Options:
  -h, --help     Print help
  -V, --version  Print version
